overlord/state/taskrunner.go

```diff
@@ -302,6 +326,22 @@ func (r *TaskRunner) Ensure() {
 	}
 }
 
+func (r *TaskRunner) cleanTask(t *Task) {
+	select {
+	case <-t.Change().Ready():
+	default:
+		return
+	}
+	cleanup, ok := r.cleanups[t.Kind()]
+	if ok {
+		if err := cleanup(t); err != nil {
```

Comment:
Good points. Will consider that further and either document or run concurrently and make it typical.